modules/eks-cluster/vpc.tf

```diff
@@ -18,6 +18,35 @@ locals {
   ]
 }
 
+data "aws_servicequotas_service_quota" "elastic_ip_quota" {
+  service_code = "ec2"
+  quota_code   = "L-0263D0A3" # Quota code for Elastic IP addresses per region
+}
+
+
+data "aws_eips" "current_usage" {}
+
+# Data source to check if the VPC exists
+data "aws_vpcs" "current_vpcs" {
+  tags = {
+    Name = local.vpc_name
+  }
+}
+
+check "elastic_ip_quota_check" {
+
+  # Only check the condition when no existing vpc is there
+  assert {
+    condition     = length(data.aws_vpcs.current_vpcs.ids) > 0 || data.aws_servicequotas_service_quota.elastic_ip_quota.value >= length(local.azs)
+    error_message = "The Elastic IP quota is insufficient to cover all local availability zones (need: ${length(local.azs)}, have: ${data.aws_servicequotas_service_quota.elastic_ip_quota.value})."
+  }
```

Comment:
as mentioned in person, would remove this assertion since it's covered in the lower one already.